src/leappto/cli.py

```diff
@@ -122,6 +140,223 @@ def _find_machine(ms, name):
+                target = source
+
+            if self.__is_int(source) and self.__is_int(target):
+                self[protocol][source] = target
+            else:
+                raise AttributeError("Source and target must be number")
+
+        def set_tcp_port(self, source, target = None):
+            self.set_port(self.PROTO_TCP, source, target)
+
+        def unset_port(self, protocol, source):
+            if not self.has_port(protocol, source):
+                raise AttributeError("Port {} is not defined".format(str(source)))
+
+            del self[protocol][source]
+
+        def unset_tcp_port(self, source):
+            self.unset_port(self.PROTO_TCP, source)
+
+        def list_ports(self, protocol):
+            return self[protocol].keys()
+
+        def list_tcp_source_ports(self):
+            return self.list_ports(self.PROTO_TCP)
+        
+        def has_port(self, protocol, source):
+            if not source in self.list_ports(protocol):
+                return False
+
+            return True
+        
+        def has_tcp_port(self, source):
+            return self.has_port(self.PROTO_TCP, source) 
+
+        def get_target_port(self, protocol, source):
+            if not self.has_port(protocol, source):
+                raise AttributeError("Port {} is not mapped".format(str(source))) 
+
+            return self[protocol][source]
+
+        def get_tcp_target_port(self, source):
+            return self.get_target_port(self.PROTO_TCP, source)
+
+        def get_protocols(self):
+            protocols = []
+            
+            if self[self.PROTO_TCP]:
+                protocols.append(self.PROTO_TCP)
+            
+            if self[self.PROTO_UDP]:
+                protocols.append(self.PROTO_UDP)
+
+            return protocols
+
+
+    def _port_scan(ip, port_range = None, shallow = False):
+        scan_args = '-sS' if shallow else '-sV'
+        
+        port_scanner = nmap.PortScanner()
+        port_scanner.scan(ip, port_range, arguments=scan_args)
+
+        ports = OrderedDict()
+
+        scan_info = port_scanner.scaninfo()
+
+        if scan_info.get('error', False):
+            raise PortScanException(scan_info['error'][0] if isinstance(scan_info['error'], list) else scan_info['error'])
+        elif ip not in port_scanner.all_hosts():
+            raise PortScanException("Machine {} not found".format(ip))
+
+        for proto in port_scanner[ip].all_protocols():
+            ports[proto] = OrderedDict()
+
+            for port in sorted(port_scanner[ip][proto]):
+                if port_scanner[ip][proto][port]['state'] != 'open':
+                    continue
+
+                ports[proto][port] = port_scanner[ip][proto][port]
+
+        return ports
+
+
+    def _port_remap(source_ports, target_ports = PortMap(), user_mapped_ports = PortMap(), user_excluded_ports = PortMap()):
+        """
+        param source_ports      - ports found by the tool on source machine
```

Comment:
Nice docstrings, but please tweak them so that they're machine readable i.e.:
`:param source_ports: description`

Rather than the current style.